### ISSUE TITLE
Recent folders menu doesn't show or takes a very long time

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/QuickFolders#284]
This may again be related to folder icons (see #283) I am seeing a number of errors related to `nsIMsgFolder.getStringProperty`:

```
11:31:19.319 Error in addSubFoldersPopupFromList
Component returned failure code: 0x80550007 [nsIMsgFolder.getStringProperty] 4 ?:4238
```

**Comment by RealRaven2000** [comment]
So the function `generateMRUList( )` now takes about 7 Seconds:

```
QuickFolders {INTERFACE,RECENTFOLDERS} 11:56:16.333  [0 ms]   
 generateMRUlist() quickfolders-util.js:1108:13
QuickFolders 11:56:23.229  [6896 ms]   
```

**Comment by RealRaven2000** [comment]
So on the one hand this function throws. But when I debug it and try again (in the exception handler) it works. I have no idea why:

![image](https://user-images.githubusercontent.com/6314674/178142952-647a41c1-6fb3-4348-b842-29c642272f0d.png)

![image](https://user-images.githubusercontent.com/6314674/178142963-8dbceef7-861b-4942-9b75-25050d1733ea.png)

Maybe @jobisoft has an idea, this is beyond my paygrade at the moment.

**Comment by RealRaven2000** [comment]
Using native code, I was able to make this faster. Apparently whenever I call nsIMsgFolder.getStringProperty("MRUTime") and error was thrown. Thunderbird core code does the same in FolderUtils.getMostRecentFolders() without that problem. using it I could speed up the history (about 250 folders) from 7500ms to 684ms. Still sluggish but better:

[QuickFolders-mx-5.11pre13.zip](https://github.com/RealRaven2000/QuickFolders/files/9079043/QuickFolders-mx-5.11pre13.zip)

To install, download zip file and drag into Add-ons Manager. This version is also compatible with Thunderbird 91 (and is a lot faster there).

**Comment by RealRaven2000** [comment]
Implemented  in 5.11 - Published 23/07/2022

I feel there is still scope for improvement here, so I am leaving this issue open. It is possible that the slow speed when clicking on the history tab to get the most recent folders is caused by internal inefficiencies in Thunderbird 102 itself. We will look out for or generate Bugzilla bugs to that end.

**Comment by RealRaven2000** [comment]
Fixed in Thunderbird 102.2 - the delay has been repaired by something in Thunderbird Core.
